### PR TITLE
Feature/bucket creation updates service #82

### PR DESCRIPTION
**Pull request by robert-clegg-tessella**
The create bucket method now takes a list of groups and a root directory as arguments. The bucket settings file is updated whenever a new bucket is created.

### CODE REVIEW

File: ditto_web_api/DittoWebApi/src/handlers/schemas/schema_helpers.py [hunk 1]
```diff
@@ -10,6 +10,23 @@ def create_object_schema_with_string_properties(list_of_properties, required=Non
...
+        "type": "object",
+        "properties": {
+            "bucket": {"type": "string"},
+            "groups": {
+                "type": "array",
+                "properties": {
+                    "items": {"type": "string"}
+                }
+            },
+            # "groups": {"type": "string"},
```
Comment (oliver-tarrant-tessella) [inline ditto_web_api/DittoWebApi/src/handlers/schemas/schema_helpers.py]: Is this comment now redundant?

Comment (robert-clegg-tessella) [reply ditto_web_api/DittoWebApi/src/handlers/schemas/schema_helpers.py]: Good spot, fixed: 0525525

File: ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py [hunk 2]
```diff
@@ -111,12 +119,34 @@ def test_post_create_bucket_successful_return_helper_coupled_with_method_schema(
     # Arguments
 
     @gen_test
-    def test_post_returns_400_when_bucket_name_is_missing(self):
-        body = {'directory': "test_dir/test_sub_dir"}
-        yield self.assert_request_returns_400_with_authorisation_okay(body)
+    def test_post_returns_403_when_bucket_name_is_missing(self):
```
Comment (oliver-tarrant-tessella) [inline ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py]: Should this be a 403?

File: ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py [hunk 3]
```diff
@@ -111,12 +119,34 @@ def test_post_create_bucket_successful_return_helper_coupled_with_method_schema(
...
+
+    @gen_test
+    def test_post_returns_400_when_groups_is_empty(self):
+        self._set_admin_user()
+        body = {'bucket': 'test-bucket', 'groups': [], 'root': '/usr/tmp/data'}
+        yield self.assert_request_returns_400_with_authorisation_okay(body)
+
+    @gen_test
+    def test_post_returns_403_when_root_is_missing(self):
+        body = {'bucket': 'test-bucket', 'groups': ['testgroup']}
```
Comment (oliver-tarrant-tessella) [inline ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py]: Should this be 403? As with test_post_returns_403_when_bucket_name_is_missing

File: ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py (hunk 2 above)
Comment (robert-clegg-tessella) [reply ditto_web_api/DittoWebApi/tests/handlers/create_bucket_handler_test.py]: Yes, I forgot to make the create bucket handler arguments required: 2dd65fb

**Pull request merged**